addons/sourcemod/scripting/freak_fortress_2.sp

```diff
@@ -6543,182 +6553,258 @@ stock GetBossIndex(client)
-				usePlayers=false;
-				switch(_operator[parentheses])
+				closeBracket = false;
+				bracket--; //We need to operate on the 1 less bracket
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=playing;
+						sum[bracket] += sum[bracket+1];
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
 					{
-						sum[parentheses]-=playing;
+						sum[bracket] -= sum[bracket+1];
+						break;
 					}
-					case 3:
+					case Operation_Multiplication:
 					{
-						sum[parentheses]*=playing;
+						sum[bracket] *= sum[bracket+1];
+						break;
 					}
-					case 4:
+					case Operation_Division:
 					{
-						sum[parentheses]/=playing;
+						sum[bracket] /= sum[bracket+1];
+						break;
 					}
-					case 5:
+					case Operation_Exponential:
 					{
-						sum[parentheses]=Pow(sum[parentheses], Float:playing);
+						sum[bracket] = Pow(sum[bracket], sum[bracket+1]);
+						break;
 					}
 					default:
 					{
-						parentheses=1;
+						//We don't got an operation for this ending bracket.(eg: "))")
+						sum[bracket] = sum[bracket+1];
 						break;
 					}
 				}
+				_operator[bracket+1] = Operation_None;
 			}
-
-			if(value[0]!='\0' && canAdd)
+			if (usePlayers)
 			{
-				canAdd=false;
-				switch(_operator[parentheses])
+				usePlayers = false;
+				//Add sum with players.
+				switch(_operator[bracket])
 				{
-					case 1:
+					case Operation_Add:
 					{
-						sum[parentheses]+=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] += float(playing);
+						break;
 					}
-					case 2:
+					case Operation_Subtract:
+					{
+						sum[bracket] -= float(playing);
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= float(playing);
+						break;
+					}
+					case Operation_Division:
 					{
-						sum[parentheses]-=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] /= float(playing);
+						break;
 					}
-					case 3:
+					case Operation_Exponential:
 					{
-						sum[parentheses]*=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] = Pow(sum[bracket], float(playing));
+						break;
 					}
-					case 4:
+					default:
 					{
-						if(!StringToFloat(value))
+						sum[bracket] = float(playing);
+						break;
+					}
+				}
+			}
+			//TODO: Add useLives back.
+			if (strValue[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
+			{
+				new Float:value = StringToFloat(strValue);
+				switch(_operator[bracket])
+				{
+					case Operation_Add:
+					{
+						sum[bracket] += value;
+						break;
+					}
+					case Operation_Subtract:
+					{
+						sum[bracket] -= value;
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= value;
+						break;
+					}
+					case Operation_Division:
+					{
+						int dotInd = StrContains(strValue, ".");
```

Comment:
Now I'm even more confused.  Can you please explain in detail why comparing StringToFloat to 0 won't work (on Steam if possible)?